Examples/Tests/particle_data_python/inputs_test_2d_particle_attr_access_picmi.py

```diff
@@ -104,24 +105,37 @@
 
 # set numpy random seed so that the particle properties generated
 # below will be reproducible from run to run
-np.random.seed(30025025)
+xp, _ = load_cupy()
+xp.random.seed(30025025)
 
 electrons = sim.particles.get("electrons")
 electrons.add_real_comp("newPid")
 
 my_id = libwarpx.amr.ParallelDescriptor.MyProc()
+num_ranks = libwarpx.amr.ParallelDescriptor.NProcs()
+
+
+def to_numpy(arr):
+    if hasattr(arr, "get"):
+        return arr.get()
+    else:
+        return np.asarray(arr)
+
+
+def nps_per_rank(rank):
+    return 10 * (rank + 1)
 
 
 def add_particles():
-    nps = 10 * (my_id + 1)
+    nps = nps_per_rank(my_id)
     x = np.linspace(0.005, 0.025, nps)
     y = np.zeros(nps)
     z = np.linspace(0.005, 0.025, nps)
     ux = np.random.normal(loc=0, scale=1e3, size=nps)
     uy = np.random.normal(loc=0, scale=1e3, size=nps)
     uz = np.random.normal(loc=0, scale=1e3, size=nps)
     w = np.ones(nps) * 2.0
-    newPid = 5.0
+    newPid = np.ones(nps) * 5.0
 
     electrons.add_particles(
```

Comment:
@ax3l @oshapoval 

I still think that the `electrons` of `electrons.add_particles(...)` is not defined within the scope of the function `add_particles` (defined on line 129 above), which I think is what pre-commit was complaining about (discussed briefly off GitHub with @oshapoval).

It looks to me like it should be either declared as `global` at the beginning of the function (i.e., on line 130) or passed as an input argument to the function, no?